File: win-prog/ProgWinVB/Dialog Boxes/DrawOrFillEllipse/DrawOrFillEllipse.vb
'--------------------------------------------------
' DrawOrFillEllipse.vb (c) 2002 by Charles Petzold
'--------------------------------------------------
Imports System
Imports System.Drawing
Imports System.Windows.Forms

Class DrawOrFillEllipse
    Inherits Form

    Private clrEllipse As Color = Color.Red
    Private bFillEllipse As Boolean = False

    Shared Sub Main()
        Application.Run(New DrawOrFillEllipse())
    End Sub

    Sub New()
        Text = "Draw or Fill Ellipse"
        ResizeRedraw = True

        Menu = New MainMenu()
        Menu.MenuItems.Add("&Options")
        Menu.MenuItems(0).MenuItems.Add("&Color...", _
                            AddressOf MenuColorOnClick)
    End Sub

    Sub MenuColorOnClick(ByVal obj As Object, ByVal ea As EventArgs)
        Dim dlg As New ColorFillDialogBox()
        dlg.Color = clrEllipse
        dlg.Fill = bFillEllipse

        If dlg.ShowDialog() = DialogResult.OK Then
            clrEllipse = dlg.Color
            bFillEllipse = dlg.Fill
            Invalidate()
        End If
    End Sub

    Protected Overrides Sub OnPaint(ByVal pea As PaintEventArgs)
        Dim grfx As Graphics = pea.Graphics
        Dim rect As New Rectangle(0, 0, ClientSize.Width - 1, _
                                        ClientSize.Height - 1)
        If bFillEllipse Then
            grfx.FillEllipse(New SolidBrush(clrEllipse), rect)
        Else
            grfx.DrawEllipse(New Pen(clrEllipse), rect)
        End If
    End Sub
End Class

File: win-prog/ProgWinCS/Dialog Boxes/DrawOrFillEllipse/DrawOrFillEllipse.cs
//------------------------------------------------
// DrawOrFillEllipse.cs © 2001 by Charles Petzold
//------------------------------------------------
using System;
using System.Drawing;
using System.Windows.Forms;

class DrawOrFillEllipse: Form
{
     Color colorEllipse = Color.Red;
     bool  bFillEllipse = false;

     public static void Main()
     {
          Application.Run(new DrawOrFillEllipse());
     }
     public DrawOrFillEllipse()
     {
          Text = "Draw or Fill Ellipse";
          ResizeRedraw = true;

          Menu = new MainMenu();
          Menu.MenuItems.Add("&Options");
          Menu.MenuItems[0].MenuItems.Add("&Color...", 
                                   new EventHandler(MenuColorOnClick));
     }
     void MenuColorOnClick(object obj, EventArgs ea)
     {
          ColorFillDialogBox dlg = new ColorFillDialogBox();

          dlg.Color = colorEllipse;
          dlg.Fill  = bFillEllipse;

          if (dlg.ShowDialog() == DialogResult.OK)
          {
               colorEllipse = dlg.Color;
               bFillEllipse = dlg.Fill;
               Invalidate();
          }
     }
     protected override void OnPaint(PaintEventArgs pea)
     {
          Graphics  grfx = pea.Graphics;
          Rectangle rect = new Rectangle(0, 0, ClientSize.Width - 1,
                                               ClientSize.Height - 1);
          if(bFillEllipse)
               grfx.FillEllipse(new SolidBrush(colorEllipse), rect);
          else
               grfx.DrawEllipse(new Pen(colorEllipse), rect);
     }
}
